### PR TITLE
Add systemd watchdog feature

### PR DESCRIPTION
**Pull request by guilhem**
:warning: May be not ready to be merged

This PR add systemd behavior for watchdog

This is a proof of concept:

- [x] Discussion may happen on way to do healthcheck (currently an `HEAD` on /ping)
- [X] PR https://github.com/coreos/go-systemd/pull/209 have to be merged

### CODE REVIEW

**Comment by emilevauge** [comment]
@containous/traefik WDYT?

**Comment by vdemeester** [comment]
Design SGTM

**Comment by errm** [comment]
LGTM, but can we get this rebased on master rather than `1.1`

**Comment by guilhem** [comment]
@errm rebased done

File: traefik.go [hunk 1]
```diff
@@ -285,10 +285,30 @@ func run(traefikConfiguration *TraefikConfiguration) {
...
+	if err != nil {
+		log.Error("Problem with watchdog", err)
+	} else if t != 0 {
+		// Send a ping each half time given
+		t = t / 2
+		go func(interval time.Duration) {
+			tick := time.Tick(interval)
+			for range tick {
+				if server.globalConfiguration.Web != nil {
+					if _, err := http.Head("http://" + server.globalConfiguration.Web.Address + "/ping"); err == nil {
```
Comment (emilevauge) [inline traefik.go]: Not sure what is the best way here...
@containous/traefik: do `/ping` and `/health` should be behind authentication or not? SSL or not?

Comment (Russell-IO) [inline traefik.go]: @emilevauge I think that /ping & /health should be on the api interface and should follow the authentication options set there (or perhaps have another config flag, defaulting back to behaviour configured on API).

Comment (trecloux) [inline traefik.go]: My 2 cents : 
* /ping and /health should be available on all configured entrypoints (So available only with SSL/TLS in some cases).
* /ping should be available without authentication, it does not carry any information.

Comment (dtomcej) [inline traefik.go]: I agree with @trecloux on this one.

Comment (SantoDE) [inline traefik.go]: -  /ping should be without authentication
-  /health should be configurable to either have authentication or not (imho)

**Comment by vdemeester** [comment]
@guilhem ping, need a rebase 👼 I can carry if you can't today 😝

File: traefik.go (hunk 1 above)
Comment (emilevauge) [inline traefik.go]: We have to deal (or not) this issue before merging. To my mind, proposed solutions are quite complex and will be difficult to implement here.
Also, I don't think using `server.globalConfiguration.Web.Address` will work here: if you have `:8080` for example.
@vdemeester @guilhem WDYT?

**Comment by emilevauge** [comment]
Rebased, but there is still https://github.com/containous/traefik/pull/818#discussion_r99114975

File: traefik.go (hunk 1 above)
Comment (guilhem) [inline traefik.go]: @emilevauge I don't have a perfect view of what will happen.
we can just remove "web" part and return a simple ping.

**Comment by guilhem** [comment]
@emilevauge can you review the latest commit, I will squash it if ok

File: traefik.go (hunk 1 above)
Comment (emilevauge) [inline traefik.go]: @guilhem Yeah too bad, but it's safer for now :'(

Comment (guilhem) [inline traefik.go]: > done is better than perfect

**Comment by emilevauge** [comment]
@guilhem seems like `make binary` is failing:
```
# github.com/containous/traefik
./traefik.go:247: too many arguments in call to daemon.SdNotify
./traefik.go:251: undefined: daemon.SdWatchdogEnabled
./traefik.go:260: too many arguments in call to daemon.SdNotify
make: *** [binary] Error 2
The command "make binary" failed and exited with 2 during .
```

**Comment by guilhem** [comment]
@emilevauge yes, it's a glide.lock problem...
I'm on it..

**Comment by vdemeester** [comment]
Small problem with the rebase, we might need to carry 😅

**Comment by SantoDE** [comment]
Sorry 👎

**Comment by vdemeester** [comment]
Carrying in #1116 😓

**Pull request closed without merging**